src/parser/csql_grammar.y

```diff
@@ -10120,6 +10126,55 @@ column_shared_constraint_def
 		DBG_PRINT}}
 	;
 
+column_on_update_def
+	: ON_ UPDATE expression_
+		{{
+			PT_NODE *attr_node;
+			PT_NODE *node = parser_new_node (this_parser, PT_ON_UPDATE);
+
+			if (node)
+			  {
+				PT_NODE *def;
+			    node->info.on_update.default_value = $3;
+			    PARSER_SAVE_ERR_CONTEXT (node, @3.buffer_pos)
+
+				def = node->info.on_update.default_value;
+			    if (def && def->node_type == PT_EXPR)
+			      {						
+					switch (def->info.expr.op)
+					  {
+					  case PT_CURRENT_TIMESTAMP:
+						node->info.on_update.default_expr_type = DB_DEFAULT_CURRENTTIMESTAMP;
+						break;
+					  case PT_CURRENT_DATE:
+						node->info.on_update.default_expr_type = DB_DEFAULT_CURRENTDATE;
+						break;
+					  case PT_CURRENT_DATETIME:
+						node->info.on_update.default_expr_type = DB_DEFAULT_CURRENTDATETIME;
+						break;
+					  case PT_SYS_TIMESTAMP:
+						node->info.on_update.default_expr_type = DB_DEFAULT_SYSTIMESTAMP;
+						break;
+					  case PT_UNIX_TIMESTAMP:
+						node->info.on_update.default_expr_type = DB_DEFAULT_UNIX_TIMESTAMP;
+						break;
+					  default:
+					    node->info.on_update.default_expr_type = DB_DEFAULT_NONE;
```

Comment:
What about SYS DATE/TIME/DATETIME?

On invalid operations, I don't think it can be accepted. A syntax error should be set and returned.